### PR TITLE
fixes ROOT file in different directory #106

### LINKED ISSUE #106: ROOT file in different directory
The current code doesn't pass the directory of the hroot file to the code that actually reads it. Therefore, it will just try to read the .root file in the current directory, not the directory of the .hroot file.

### PR DESCRIPTION
**Pull request by NikEfth**
Introduced virtual InputStreamFromROOTFile::set_up(const std::string&). The function takes over functionality previously in post_processing(). But in addition the full path of the header file location can be used. 

FilePath is used for the merging of the strings. As this is the first time the class is used small modifications had to be made to make the functionality relevant. 

1. new function get_as_string() const; which returns the current my_string, wether it is path or file. 
2. If my_string has just a filename, function get_path_only) returns an empty string.

### CODE REVIEW

**Comment by NikEfth** [comment]
This check boxes will close #96 
- [ ] Header format is only partially documented (not for ECAT for instance)
- [ ] InputStreamFromROOT constructors don't see all fields (should also do stuff from post_processing(), but that's a virtual function so cannot call this from a constructor)
- [ ] parsing of user-defined scanner has no checks nor defaults
- [ ] Issue #95

File: src/IO/InputStreamFromROOTFileForCylindricalPET.cxx [hunk 1]
```diff
@@ -155,8 +155,15 @@ InputStreamFromROOTFileForCylindricalPET::initialise_keymap()
 bool InputStreamFromROOTFileForCylindricalPET::
 post_processing()
 {
-    if (base_type::post_processing())
-        return true;
+    return false;
```
Comment (KrisThielemans) [inline src/IO/InputStreamFromROOTFileForCylindricalPET.cxx]: safer to keep these lines as they were. you never know if `base_type::post_processing` will do something

File: src/IO/InputStreamFromROOTFile.cxx [hunk 2]
```diff
@@ -63,16 +64,28 @@ InputStreamFromROOTFile::initialise_keymap()
...
+
+Succeeded
+InputStreamFromROOTFile::set_up(const std::string & header_path)
 {
     // Read the 4 bytes to check whether this is a ROOT file, indeed.
-    FileSignature signature(filename.c_str());
+    FilePath f(filename,false);
+    f.prepend_directory_name(header_path);
+
+    std::string fullfilename = f.get_as_string();
```
Comment (KrisThielemans) [inline src/IO/InputStreamFromROOTFile.cxx]: could be `const std::string` I guess

File: src/listmode_buildblock/CListModeDataROOT.cxx [hunk 3]
```diff
@@ -60,6 +61,9 @@ CListModeDataROOT(const std::string& hroot_filename)
     if(!this->parser.parse(hroot_filename.c_str()))
         error("CListModeDataROOT: error parsing '%s'", hroot_filename.c_str());
 
+    FilePath f(hroot_filename);
+    root_file_sptr->set_up( f.get_path_only());
```
Comment (KrisThielemans) [inline src/listmode_buildblock/CListModeDataROOT.cxx]: should check return value

File: src/IO/InputStreamFromROOTFile.cxx (hunk 2 above)
Comment (NikEfth) [reply src/IO/InputStreamFromROOTFile.cxx]: :ok:

File: src/IO/InputStreamFromROOTFileForCylindricalPET.cxx (hunk 1 above)
Comment (NikEfth) [reply src/IO/InputStreamFromROOTFileForCylindricalPET.cxx]: :ok:

File: src/listmode_buildblock/CListModeDataROOT.cxx (hunk 3 above)
Comment (NikEfth) [reply src/listmode_buildblock/CListModeDataROOT.cxx]: :ok:

**Pull request merged**